lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -243,162 +223,281 @@ internal static void ReadCategory(XmlReader reader, SyndicationCategory category
-            }
-            else if (reader.IsStartElement(Atom10Constants.UpdatedTag, Atom10Constants.Atom10Namespace))
-            {
-                reader.ReadStartElement();
-                result.LastUpdatedTime = DateFromString(reader.ReadString(), reader);
-                reader.ReadEndElement();
-            }
-            else
+
+            string name = reader.LocalName;
+            string ns = reader.NamespaceURI;
+
+            if(ns == Atom10Constants.Atom10Namespace)
             {
-                return false;
-            }
-            return true;
+                switch (name)
+                {
+                    case Atom10Constants.AuthorTag:
+                        result.Authors.Add(await ReadPersonFromAsync(reader, result));
+                        break;
+                    case Atom10Constants.CategoryTag:
+                        result.Categories.Add(await ReadCategoryFromAsync(reader, result));
+                        break;
+                    case Atom10Constants.ContributorTag:
+                        result.Contributors.Add(await ReadPersonFromAsync(reader, result));
+                        break;
+                    case Atom10Constants.GeneratorTag:
+                        result.Generator = await reader.ReadElementStringAsync();
+                        break;
+                    case Atom10Constants.IdTag:
+                        result.Id = await reader.ReadElementStringAsync();
+                        break;
+                    case Atom10Constants.LinkTag:
+                        result.Links.Add(await ReadLinkFromAsync(reader, result));
+                        break;
+                    case Atom10Constants.LogoTag:
+                        result.ImageUrl = new Uri(await reader.ReadElementStringAsync(), UriKind.RelativeOrAbsolute);
+                        break;
+                    case Atom10Constants.RightsTag:
+                        result.Copyright = await ReadTextContentFromAsync(reader, "//atom:feed/atom:rights[@type]");
+                        break;
+                    case Atom10Constants.SubtitleTag:
+                        result.Description = await ReadTextContentFromAsync(reader, "//atom:feed/atom:subtitle[@type]");
+                        break;
+                    case Atom10Constants.TitleTag:
+                        result.Title = await ReadTextContentFromAsync(reader, "//atom:feed/atom:title[@type]");
+                        break;
+                    case Atom10Constants.UpdatedTag:
+                        await reader.ReadStartElementAsync();
+                        result.LastUpdatedTime = DateFromString(await reader.ReadStringAsync(), reader);
+                        await reader.ReadEndElementAsync();
+                        break;
+
+
+                    case Atom10Constants.IconTag:
+                        result.IconImage = await reader.ReadElementStringAsync();
+                        break;
```

Comment:
Atom spec says icon is of type atomUri. Should this be a string or a Uri? Do feeds ever put something other than a Uri here?